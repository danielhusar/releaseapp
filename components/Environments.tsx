import React from 'react'
import { Box, Heading } from '@chakra-ui/core'
import Table from './Table'
import EnvironmentRow from './EnvironmentRow'
import { IEnvironment } from '~/@types/environment'

export interface IEnvironments {
  environments: IEnvironment[]
}

export default function Environments({ environments }: IEnvironments) {
  return (
    <Box
      p="6"
      borderWidth="1px"
      background="white"
      overflow="hidden"
      display="flex"
      flexDirection="column"
      data-testid="environment"
    >
      <Heading size="md" fontWeight={400} marginBottom={5}>
        Environments
      </Heading>

      <Table head={['Name', 'Status', 'Date', 'Build', 'Commit SHA', 'Actions']}>
        {environments.map((environment) => (
          <EnvironmentRow {...environment} key={environment.id} />
        ))}
      </Table>
    </Box>
  )
}
