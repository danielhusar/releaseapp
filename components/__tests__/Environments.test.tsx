import React from 'react'
import { render, app } from './helpers'
import Environments from '../Environments'

const RealDate = Date.now

beforeAll(() => {
  global.Date.now = jest.fn(() => new Date('2020-04-20T15:48:38.039Z').getTime())
})

afterAll(() => {
  global.Date.now = RealDate
})

describe('Environments', () => {
  it('renders', () => {
    const { container } = render(<Environments {...app} />)
    expect(container).toMatchSnapshot()
  })
})
