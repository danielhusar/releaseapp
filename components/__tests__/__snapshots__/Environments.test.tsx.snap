// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Environments renders 1`] = `
.emotion-7 {
  padding: 1.5rem;
  border-width: 1px;
  background: white;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0 {
  font-size: 1.25rem;
  line-height: 1.25;
  font-weight: 400;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  margin-bottom: 1.25rem;
}

.emotion-6 {
  border: none;
  text-align: left;
  font-size: 0.875rem;
  width: 100%;
}

.emotion-6 th {
  padding: 20px 5px;
  font-weight: 500;
}

.emotion-6 td {
  border-top: 1px solid #ccc;
  padding: 30px 5px;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  font-size: calc(2rem / 2.5);
  line-height: 2rem;
  vertical-align: top;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-radius: 9999px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  width: 2rem;
  height: 2rem;
  background-color: #A0AEC0;
  color: #fff;
  margin-right: 1rem;
}

.emotion-1 {
  width: 100%;
  height: 100%;
}

.emotion-4 {
  display: inline-block;
  padding-left: 0.25rem;
  padding-right: 0.25rem;
  text-transform: uppercase;
  font-size: 0.75rem;
  border-radius: 0.125rem;
  font-weight: 700;
  white-space: nowrap;
  vertical-align: middle;
  background-color: #38a169;
  color: #fff;
}

.emotion-5 {
  border-radius: 0.25rem;
  font-weight: 600;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 250ms;
  transition: all 250ms;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  position: relative;
  white-space: nowrap;
  vertical-align: middle;
  line-height: 1.2;
  outline: none;
  height: 2.5rem;
  min-width: 2.5rem;
  font-size: 1rem;
  padding-left: 1rem;
  padding-right: 1rem;
  background-color: #3182ce;
  color: #fff;
}

.emotion-5:hover {
  background-color: #2a69ac;
}

.emotion-5:focus {
  box-shadow: 0 0 0 3px rgba(66,153,225,0.6);
}

.emotion-5:active,
.emotion-5[data-active=true] {
  background-color: #1e4e8c;
}

.emotion-5:disabled,
.emotion-5:disabled:focus,
.emotion-5:disabled:hover,
.emotion-5[aria-disabled=true],
.emotion-5[aria-disabled=true]:focus,
.emotion-5[aria-disabled=true]:hover {
  opacity: 0.4;
  cursor: not-allowed;
  box-shadow: none;
}

<div>
  <div
    class="emotion-7"
    data-testid="environment"
  >
    <h2
      class="emotion-0"
    >
      Environments
    </h2>
    <table
      class="emotion-6"
    >
      <thead>
        <tr>
          <th>
            Name
          </th>
          <th>
            Status
          </th>
          <th>
            Date
          </th>
          <th>
            Build
          </th>
          <th>
            Commit SHA
          </th>
          <th>
            Actions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <div
              class="emotion-3"
            >
              <div
                class="emotion-2"
              >
                <div
                  class="emotion-1"
                >
                  <svg
                    fill="#fff"
                    role="img"
                    viewBox="0 0 128 128"
                  >
                    <g>
                      <path
                        d="M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
                      />
                      <path
                        d="M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
                      />
                    </g>
                  </svg>
                </div>
              </div>
              <a
                href="/app/291/environment/6701"
              >
                staging
              </a>
            </div>
          </td>
          <td>
            <div
              class="emotion-4"
            >
              done
            </div>
          </td>
          <td>
            2 weeks ago
          </td>
          <td>
            10758
          </td>
          <td>
            <a
              href="https://github.com/tmcclung/release-rails-starter/commit/7ca25300fd20b337d06b087a15ed4f856e23454d"
              rel="noreferrer"
              target="_blank"
            >
              7ca2530
            </a>
          </td>
          <td>
            <a
              href="/app/291/environment/6701"
            >
              <button
                class="emotion-5"
                type="button"
              >
                View
              </button>
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
